src/main/java/io/confluent/connect/jdbc/sink/metadata/FieldsMetadata.java

```diff
@@ -129,7 +130,15 @@ public static FieldsMetadata extract(
       );
     }
 
-    return new FieldsMetadata(keyFieldNames, nonKeyFieldNames, allFields);
+    final Map<String, SinkRecordField> allFieldsOrdered = new LinkedHashMap<>();
```

Comment:
By separating allFields and allFieldsOrdered, you're removing the PK-first ordering of the columns in favor of whatever order the record specifies them in. I think this might violate some assumptions that we have downstream.

I'm also concerned about what happens when allFields contains a key that's not in the record (such as topic/partition/offset), since these are added in a previous step to allFields, but never appear in allFieldsOrdered.

I think the build failures may be linked to this change in semantics.

Reply:
Ok, then perhaps this is not the right approach, since the previous method will not preserve original column ordering in the destination table (as PK's will appear first).

Perhaps the approach needed is for the query builder to not use the order from allFields :/

Reply:
I have committed additional code to make sure all fields are added in a determinate order to the `allFieldsOrdered` map.

Looks like tests are completing now.